### ISSUE TITLE
v4.2.0 throws a missing module error for `fs`

### ISSUE THREAD

**Issue by mfts** [tus/tus-js-client#715]
**Describe the bug**

missing module `fs`

```sh
Module not found: Can't resolve 'fs'

https://nextjs.org/docs/messages/module-not-found

Import trace for requested module:
./node_modules/proper-lockfile/lib/adapter.js
./node_modules/proper-lockfile/index.js
./node_modules/tus-js-client/lib.esm/node/urlStorage.js
./node_modules/tus-js-client/lib.esm/node/index.js
./lib/files/tus-upload.ts
./components/upload-zone.tsx
./components/documents/documents-list.tsx
./pages/documents/index.tsx
 ⨯ ./node_modules/graceful-fs/graceful-fs.js:1:1
 ```

**To Reproduce**

1. update package
2. `npm run dev`
3. throws error in browser

**Expected behavior**
Not throw an error

**Setup details**
Setup is next.js@14.2.7

- Runtime environment: browser
- Used tus-js-client version: 4.2.0
- Used tus server software: 1.8.0

**Comment by SanderMuller** [comment]
This is also a breaking change for us (using webpack 5.94.0 via Laravel Mix)

**Comment by Acconut** [comment]
Can you provide a minimal example to reproduce this? Did it work in 4.1.0? From the [changes](https://github.com/tus/tus-js-client/compare/v4.1.0...v4.2.0) between the two releases, I don't see a direct reason why it would now pick up the Node.js code (`lib.esm/node/urlStorage.js`) versus the browser code (`lib.esm/browser/urlStorage.js`).

**Comment by Acconut** [comment]
I cannot reproduce this with Next.js. Created a new app with `npx create-next-app@latest` and `npm install tus-js-client@4.2.0` and added this to `page.js`:

```js
import * as tus from "tus-js-client";

export default function Home() {
  return (
          <div>Is supported: {tus.isSupported.toString()}</div>
  )
}
```

That works as expected without an error:

![image](https://github.com/user-attachments/assets/db6230aa-9a2a-4a6c-b56b-6e43408a59ed)

Please provide more details to reproduce this. Or better, a complete, minimal example.

**Comment by Acconut** [comment]
[v4.2.2](https://github.com/tus/tus-js-client/releases/tag/v4.2.2) has been released and drops `node:`. Please try it out and let us know if this helps.

Even if it solves the problem, it raises the question why Webpack picks up Node.js-only files when building for a browser. Maybe it has always done this while polyfilling Node.js modules, but now it cannot polyfill with the `node:` protcol and thus fails now. However, it shouldn't inspect Node.js-only files in the first place.

**Comment by Acconut** [comment]
@aaronforloop Thanks, I can reproduce it with your repository. I'll look into it.

**Comment by Acconut** [comment]
I might have found the problem and it's not visible by looking at the changes in this repository. Instead, the v4.2.0 release on NPM accidentally includes `.d.ts` and other files from #683 (see https://www.npmjs.com/package/tus-js-client/v/4.2.0?activeTab=code), that slipped in when I do releases on my computer:

![image](https://github.com/user-attachments/assets/4e3ba511-ae58-4656-a164-e0c049996007)

This PR is an attempt to move tus-js-client to TypeScript, but hasn't been merged, so its files shouldn't be in the release. For some reason unknown to me these files cause Webpack to fail.

I just released v4.2.3 to NPM without these problematic, unintended files. Please test it and let me know if that fixes the problem.

To prevent this issue in the future, I will improve the build script to clean unwanted files and also work on automating releases on CI versus doing them manually. Apologies for that.

**Comment by aaronforloop** [comment]
Thank you for the prompt fix! We've tested v4.2.3, and it resolves the issue on our end. Appreciate your efforts.

**Comment by Acconut** [comment]
Thanks for the assistance, without which it would have been a lot hard to debug this! I will close this new. Feel free to leave a comment if you are still struggling with this specific Webpack error.
